docs/storage.md

```diff
@@ -409,4 +415,36 @@ This is mainly useful for testing and demos.
 type: FILESYSTEM
 config:
   directory: ""
+prefix: ""
+```
+
+## Prefix
+
+Now Thanos also supports multi-tenancy at the object storage using custom prefix. Check [this](https://github.com/thanos-io/thanos/issues/1318) issue. This feature is implemented in such a way that, no client level changes are required. All object storage implementations support this `prefix` key in their object storage config yaml.
```

Comment:
It would be definitely nice to mention `WHY` someone would use this field.

I would NOT mention multi-tenancy use case, let's talk on issue more to clarify the purpose, will add comment on https://github.com/thanos-io/thanos/issues/1318

Reply:
Still to do (:

Reply:
Sorry for the delay, I didn't get chance to work on this for some other priority. Should be done by tomorrow..

Reply:
Yes, `Prefix: "/tenant-0"` should work too.

Reply:
Updated the docs in the latest commit